fjord/suggest/providers/trigger/models.py

```diff
@@ -0,0 +1,253 @@
+            u'locale-independent if possible.'
+        )
+    )
+    sortorder = models.IntegerField(
+        default=5,
+        help_text=(
+            u'RequireD: Allows you to dictate which trigger rules are more '
+            u'important and thus show up first in a list of trigger rules '
+            u'suggestions.'
+        )
+    )
+    is_enabled = models.BooleanField(default=False)
+
+    # Trigger filters
+    locales = models.CharField(
+        blank=True,
+        max_length=255,
+        help_text=(
+            u'Comma-separated list of locale values to match or empty to '
+            u'match all locales.'
+        )
+    )
+    products = models.ManyToManyField(
+        Product,
+        blank=True,
+        help_text=(
+            u'List of products to match or none to match all feedback.'
+        )
+    )
+    keywords = models.TextField(
+        blank=True,
+        help_text=(
+            u'Key words and phrases (in quotes) each on a separate line '
+            u'or empty to match all feedback.'
+        )
+    )
+    versions = models.CharField(
+        blank=True,
+        max_length=255,
+        help_text=(
+            u'Comma delimited list of versions to match or empty to '
+            u'match all versions. Use * at the end to match anything after '
+            u'that point. For example "38*" matches all version 38 releases.'
+        )
+    )
+    url_exists = models.NullBooleanField(
+        default=None,
+        blank=True,
+        help_text=u'Has a url, does not have a url or do not care.'
+    )
+
+    objects = TriggerRuleManager()
+
+    def save(self, *args, **kwargs):
+        self.keywords = self.clean_keywords(self.keywords)
+        self.locales = self.clean_locales(self.locales)
+        self.versions = self.clean_versions(self.versions)
+        super(TriggerRule, self).save(*args, **kwargs)
+
+    @classmethod
+    def clean_keywords(cls, keywords):
+        """Clean up whitespace in keywords"""
+        lines = keywords.splitlines()
+        return '\n'.join([line.strip() for line in lines if line.strip()])
+
+    @classmethod
+    def clean_locales(cls, locales):
+        """Cleans up whitespace and commas in locales"""
+        return ','.join([
+            locale.strip()
+            for locale in locales.split(',')
+            if locale.strip()
+        ])
+
+    @classmethod
+    def clean_versions(cls, versions):
+        """Cleans up whitespace in versions"""
+        return ','.join([
+            version.strip()
+            for version in versions.split(',')
+            if version.strip()
+        ])
+
+    def match_locale(self, locale):
+        # Check if this rule matches all locales
+        if not self.locales:
+            return True
+
+        # Check if there was a locale supplied
+        if not locale:
+            return False
+
+        # Check if the locale was in the rule
+        return locale in self.locales.split(',')
```

Comment:
If you took the behavior of the `locales` field, and made it into its own special kind of field that stored multiple values as a comma-delimited string, converting to a string when it goes into the database, and converting into a list when it comes out, I'd be quite happy, and I would steal it to use on Edwin. I'm doing a similar thing there with ad-hoc methods like this.

That would be hard though, and the ad-hoc way works fine.